### PR TITLE
Soften error to warning in template processor when fails to read file

### PR DESCRIPTION
**Pull request by sfc-gh-pjafari**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Template processor should not fail if it can't read a file.

### CODE REVIEW

File: src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py [hunk 1]
```diff
@@ -58,38 +58,45 @@ def expand_templates_in_file(
...
+                        raise InvalidTemplateInFileError(
+                            src_file_name, e, e.lineno
+                        ) from e
+
+                    except jinja2.UndefinedError as e:
+                        raise InvalidTemplateInFileError(src_file_name, e) from e
+
+                    if expanded_template != file.contents:
+                        file.edited_contents = expanded_template
+        except UnicodeDecodeError as err:
```
Comment (sfc-gh-melnacouzi) [inline src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py:96]: why would we get this error?

Comment (sfc-gh-melnacouzi) [reply src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py:96]: I guess it makes sense for binary files.

**Review by sfc-gh-melnacouzi — DISMISSED** [review]

**Review by sfc-gh-bdufour — COMMENTED** [review]
No test?

File: tests/nativeapp/codegen/templating/test_templates_processor.py [hunk 2]
```diff
@@ -213,3 +217,24 @@ def test_file_with_undefined_variable():
...
+@mock.patch(CLI_GLOBAL_TEMPLATE_CONTEXT, {})
+@mock.patch(f"{TEMPLATE_PROCESSOR}.cc.warning")
+def test_expand_templates_in_file_unicode_decode_error(mock_cc_warning):
+    file_name = ["test_file.txt"]
+    file_contents = ["This is a test file"]
+    with TemporaryDirectory() as tmp_dir:
+        bundle_result = bundle_files(tmp_dir, file_name, file_contents)
+        templates_processor = TemplatesProcessor(bundle_ctx=bundle_result.bundle_ctx)
+        with mock.patch(
+            f"{ARTIFACT_PROCESSOR}.ProjectFileContextManager.__enter__",
```
Comment (sfc-gh-bdufour) [inline tests/nativeapp/codegen/templating/test_templates_processor.py:231]: why not just mock `edit_file`? That's the public contract, whereas this is an implementation detail.

File: src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py (hunk 1 above)
Comment (sfc-gh-mchok) [reply src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py:96]: So that's where those errors are coming from 🤯

Comment (sfc-gh-pjafari) [reply src/snowflake/cli/_plugins/nativeapp/codegen/templates/templates_processor.py:96]: @sfc-gh-mchok you saw these on the dashboard?! Yeah our template processor was dying every time there was a file it couldn't read :D

**Review by sfc-gh-melnacouzi — APPROVED** [review]

**Pull request merged**